service/controller/v1/resource/configmap/desired.go

```diff
@@ -59,11 +60,24 @@ func (r *Resource) getDesiredState(ctx context.Context) (*corev1.ConfigMap, erro
 		return nil, microerror.Mask(err)
 	}
 
+	// Prometheus YAML marshalling obscures remote write passwords for security,
+	// so we store the Cortex password, and write it back after marshalling.
+	var remoteWritePassword string
+	if len(newPrometheusConfig.RemoteWriteConfigs) == 1 {
```

Comment:
Assuming we will need only 1 config.

Reply:
Yeah, I think that's a fine assumption for now